### ISSUE TITLE
Lack of proper documentation

### ISSUE THREAD

**Issue by ar3s3ru** [samsarahq/thunder#303]
From the README, the library seems very promising and powerful.

But looking at the [documentation](https://godoc.org/github.com/samsarahq/thunder), there's almost none: this makes the learning curve steeper and harder for other developers to choose to adopt this library.

**Comment by kevinmichaelchen** [comment]
Agreed. I think there should at least be a Gitter or Discord or something to build a community and knowledge base.

I'm struggling to get optional fields working right now, and don't really know where to look.

```go
Title        string   `graphql:",optional"`
```
